ckanext/cfpb_extrafields/plugin.py

```diff
@@ -75,30 +75,49 @@ def _redirect_to_edit_on_change(self, resource, field):
 
     def _delete_and_rebuild_datadict(self, resource):
         import json
+        import unicodedata
+
         if 'datadict' in resource and 'id' in resource:
-            record = resource['datadict']
-            resource.pop('datadict')
-            json_record = json.loads(record)
+            record = resource.pop('datadict')
+
+            if record:
+                # Cleanse of errant unicode characters
+                record = unicodedata.normalize('NFKD', record).encode('ascii', 'ignore')
```

Comment:
It normalizes non-ascii characters ot their ascii equivalent, if one exists, so formatting should be kept.  As far as I can tell, this fixes the long entry problem as well, I created one with 500 or so words that worked fine.  New lines are still not being retained though, but that was existing behavior.  Let me know if you know where that is happening.